persistence/src/commonMain/kotlin/com/wire/kalium/persistence/backup/DatabaseImporter.kt

```diff
@@ -56,6 +50,41 @@ class DatabaseImporterImpl(private val sqlDriver: SqlDriver) : DatabaseImporter
         restoreTable("MessageRestrictedAssetContent")
     }
 
+    private fun restoreConversations() {
+        // Before restoring any conversations, we need to set the last_read_date of the backup conversations to the last_read_date of the
+        // current conversations if it is more recent that what the backup states
+
+        /*
+        Parsing Ambiguity
+        When the INSERT statement to which the UPSERT is attached takes its values from a SELECT statement,
+        there is a potential parsing ambiguity. The parser might not be able to tell if the "ON" keyword is
+        introducing the UPSERT or if it is the ON clause of a join. To work around this, the SELECT statement
+        should always include a WHERE clause, even if that WHERE clause is just "WHERE true".
+        Ambiguous use of ON:
+        INSERT INTO t1 SELECT * FROM t2
+        ON CONFLICT(x) DO UPDATE SET y=excluded.y;
+
+        Ambiguity resolved using a WHERE clause:
+        INSERT INTO t1 SELECT * FROM t2 WHERE true
+        ON CONFLICT(x) DO UPDATE SET y=excluded.y;
+
+        https://www.sqlite.org/lang_UPSERT.html
+         */
+
+        sqlDriver.execute(
```

Comment:
I think we need to do the same with the last_notified_date

Reply:
What do you mean? we are handling those ones on line 83